polyfill/lib/calendar.mjs

```diff
@@ -633,18 +633,20 @@ const nonIsoHelperBase = {
   adjustCalendarDate(calendarDate, cache, overflow /*, fromLegacyDate = false */) {
     if (this.calendarType === 'lunisolar') throw new RangeError('Override required for lunisolar calendars');
     this.validateCalendarDate(calendarDate);
-    const largestMonth = this.monthsInYear(calendarDate, cache);
-    let { month, year, eraYear, monthCode } = calendarDate;
-
     // For calendars that always use the same era, set it here so that derived
     // calendars won't need to implement this method simply to set the era.
     if (this.constantEra) {
       // year and eraYear always match when there's only one possible era
-      if (year === undefined) year = eraYear;
-      if (eraYear === undefined) eraYear = year;
-      calendarDate = { ...calendarDate, era: this.constantEra, year, eraYear };
+      const { year, eraYear } = calendarDate;
```

Comment:
This refactor was needed on the TS side to reduce type casts needed when `let` variables were mutated. I thought it made the code easier to follow in JS so I left it in. Happy to revert if desired.